src/value.c

```diff
@@ -787,6 +787,40 @@ static json_t *json_string_copy(const json_t *string)
     return json_stringn_nocheck(s->value, s->length);
 }
 
+static json_t *json_vsprintf(const char *fmt, va_list ap) {
+    int length;
+    char *buf;
+    json_t *result;
+
+    va_list aq;
+    va_copy(aq, ap);
+
+    length = vsnprintf(NULL, 0, fmt, ap);
+    if (length == 0)
+        return json_string("");
+
+    buf = jsonp_malloc(length + 1);
+    if (!buf)
+        return NULL;
+
+    vsnprintf(buf, length + 1, fmt, aq);
+    result = json_stringn(buf, length);
```

Comment:
An extra allocation/string copy could be avoided if we directly called `utf8_check_string` then use `jsonp_stringn_nocheck_own`.

Your call if you want to do take this suggestion or ignore it.

Reply:
Fixed, good catch!